pydantic/types.py

```diff
@@ -268,11 +268,18 @@ class PyObject:
 
     @classmethod
     def __get_validators__(cls) -> 'CallableGenerator':
-        yield str_validator
         yield cls.validate
 
     @classmethod
     def validate(cls, value: Any) -> Any:
+        if isinstance(value, Callable):  # type: ignore
+            return value
+
+        try:
+            value = str_validator(value)
+        except errors.StrError:
+            raise errors.PyObjectCommonError()
```

Comment:
```suggestion
            raise errors.PyObjectError(error_message='value is neither a valid import path not a valid callable')
```

I think this would be clearer, then you can remove `PyObjectCommonError` completely.

Sorry not to be more clear before.

Reply:
Oh, I see now. If changing error messages is expected by users, then I'll fix it soon